packages/core/src/components/Form.tsx

```diff
@@ -550,19 +558,21 @@ export default class Form<
    */
   onChange = (formData: T | undefined, newErrorSchema?: ErrorSchema<T>, id?: string) => {
     const { extraErrors, omitExtraData, liveOmit, noValidate, liveValidate, onChange } = this.props;
-    const { schemaUtils, schema } = this.state;
+    const { schemaUtils, schema, retrievedSchema } = this.state;
+
     if (isObject(formData) || Array.isArray(formData)) {
-      const newState = this.getStateFromProps(this.props, formData);
+      const newState = this.getStateFromProps(this.props, formData, retrievedSchema);
       formData = newState.formData;
     }
 
     const mustValidate = !noValidate && liveValidate;
     let state: Partial<FormState<T, S, F>> = { formData, schema };
     let newFormData = formData;
 
+    let _retrievedSchema: S | undefined;
     if (omitExtraData === true && liveOmit === true) {
-      const retrievedSchema = schemaUtils.retrieveSchema(schema, formData);
-      const pathSchema = schemaUtils.toPathSchema(retrievedSchema, '', formData);
+      _retrievedSchema = schemaUtils.retrieveSchema(schema, formData);
```

Comment:
Would this work?
```suggestion
      _retrievedSchema = newState.retrievedSchema ?? schemaUtils.retrieveSchema(schema, formData);
```

Reply:
When I run the test, it fails on some tests with `{"omitExtraData":true,"liveOmit":true}`. Not fully sure, but probably, because the schema is old with unnecessary properties. 4 tests failed